yaptide/batch/string_templates.py

```diff
@@ -82,9 +87,104 @@
     wait # wait for all children, this is important!
 }}
 
+FILE_TO_WATCH=$WORK_DIR/shieldhit_`printf %04d $SLURM_ARRAY_TASK_ID`.log
+srun python3 $WATCHER_FILE --filepath=$FILE_TO_WATCH\\
+    --job_id=$SLURM_JOB_ID --task_id=$SLURM_ARRAY_TASK_ID &
+
 trap 'sig_handler' SIGUSR1
 
 # execute simulation
 srun shieldhit --beamfile=$BEAM_FILE --geofile=$GEO_FILE --matfile=$MAT_FILE --detectfile=$DETECT_FILE\\
-    -n {particle_no} -N $RNG_SEED  $WORK_DIR
+    -n {particle_no} -N $RNG_SEED  $WORK_DIR &
+
+wait
+"""  # skipcq: FLK-E501
+
+PYTHON_WATCHER_SCRIPT: str = """from pathlib import Path
+import re
+import time
+import argparse
+import signal
+
+def log_generator(thefile):
+    while True:
+        line = thefile.readline()
+        if not line:
+            time.sleep(1)
+            continue
+        yield line
+
+
+def read_file(filepath: Path, job_id: str, task_id: int):
+    run_match = r"\\bPrimary particle no.\\s*\\d*\\s*ETR:\\s*\\d*\\s*hour.*\\d*\\s*minute.*\\d*\\s*second.*\\b"
+    complete_match = r"\\bRun time:\\s*\\d*\\s*hour.*\\d*\\s*minute.*\\d*\\s*second.*\\b"
+    requested_match = r"\\bRequested number of primaries NSTAT"
+
+    logfile = None
+    for _ in range(30):  # 30 stands for maximum attempts
+        try:
+            logfile = open(filepath)  # skipcq: PTC-W6004
+            break
+        except FileNotFoundError:
+            time.sleep(3)
+
+    if logfile is None:
+        up_dict = {
+            "task_state": "FAILED"
+        }
+        print(f"Update for task: {{task_id}} - FAILED")
+        return
+
+    loglines = log_generator(logfile)
+    for line in loglines:
+        if re.search(run_match, line):
+            splitted = line.split()
+            up_dict = {
+                "simulated_primaries": int(splitted[3]),
+                "estimated_time": {
+                    "hours": int(splitted[5]),
+                    "minutes": int(splitted[7]),
+                    "seconds": int(splitted[9]),
+                }
+            }
+            print(f"Update for task: {{task_id}} - simulated primaries: {{splitted[3]}}")
+
+        elif re.search(requested_match, line):
+            splitted = line.split(": ")
+            up_dict = {
+                "simulated_primaries": 0,
+                "requested_primaries": int(splitted[1]),
+                "task_state": "RUNNING"
+            }
+            print(f"Update for task: {{task_id}} - RUNNING")
+
+        elif re.search(complete_match, line):
+            splitted = line.split()
+            up_dict = {
+                "run_time": {
+                    "hours": int(splitted[2]),
+                    "minutes": int(splitted[4]),
+                    "seconds": int(splitted[6]),
+                },
+                "task_state": "COMPLETED"
+            }
+            print(f"Update for task: {{task_id}} - COMPLETED")
+            return
+
+
+if __name__ == '__main__':
+    signal.signal(signal.SIGUSR1, signal.SIG_IGN)
```

Comment:
why do you need to call `signal` ?

Reply:
I wanted to make sure, SIGUSR1 is ignored by python script

Reply:
Ah wait I didn't understand the question. This is just registering handler for SIGUSR1 as SIG_IGN